nsot/util.py

```diff
@@ -44,3 +47,53 @@ def get_template_env(package="nsot", extra_filters=None, extra_globals=None):
     env.globals.update(j_globals)
 
     return env
+
+
+#: Namedtuple for resultant items from ``parse_set_query()``
+SetQuery = collections.namedtuple('SetQuery', 'action name value')
+
+
+def parse_set_query(query):
+    """
+    Parse a representation of set operations for attribute/value pairs into
+    (action, name, value) and return a list of ``SetQuery`` objects.
+
+    Computes left-to-right evaluation, where the first character indicates the
+    set operation:
+
+    + "+" indicates a union
+    + "-" indicates a difference
+    + no marker indicates an intersection
+
+    For example::
+
+        >>> parse_set_query('+owner=team-networking')
+        [SetQuery(action='union', name='owner', value='team-networking')]
+        >>> parse_set_query('foo=bar')
+        [SetQuery(action='intersection', name='foo', value='bar')]
+        >>> parse_set_query('foo=bar -owner=team-networking')
+        [SetQuery(action='intersection', name='foo', value='bar'),
+         SetQuery(action='difference', name='owner', value='team-networking')]
+
+    :param query:
+        Set query string
+    """
+    log.debug('Incoming query = %r' % (query,))
+    queries = query.split(' ')
```

Comment:
I'm concerned that this would split strangely for 

`"foo=bar        spam=egg"`

where there is more than one space between each attribute pair. The default whitespace splitting has pretty different behavior, where it collapses all whitespace. Any opinion here?